src/bridge/cockpitfsread.c

```diff
@@ -238,19 +239,34 @@ cockpit_fsread_prepare (CockpitChannel *channel)
     }
 
   mapped = g_mapped_file_new_from_fd (self->fd, FALSE, &error);
+  if (mapped)
+    {
+      bytes = g_mapped_file_get_bytes (mapped);
+    }
+  else if (g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NODEV))
+    {
+      gchar *contents;
+      gsize length;
+
+      g_clear_error (&error);
+      if (g_file_get_contents (self->path, &contents, &length, &error))
+        bytes = g_bytes_new (contents, length);
```

Comment:
This is not very efficient, in certain corner cases, but pragmatically solves the actual issue. I think this is good enough until someone requires that those corner cases be fixed.